presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -312,6 +324,17 @@ private SubPlan buildFragment(PlanNode root, FragmentProperties properties, Plan
 
             Map<Symbol, Type> fragmentSymbolTypes = filterKeys(symbolAllocator.getTypes().allTypes(), in(extractOutputSymbols(root)));
             planSanityChecker.validatePlanFragment(root, session, metadata, sqlParser, TypeProvider.viewOf(fragmentSymbolTypes), warningCollector);
+
+            Set<PlanNodeId> tableWriterNodeIds = getTableWriterNodeIds(root);
+            boolean outputTableWriterFragment = tableWriterNodeIds.stream().anyMatch(outputTableWriterNodeIds::contains);
+            if (outputTableWriterFragment) {
+                verify(
+                        outputTableWriterNodeIds.containsAll(tableWriterNodeIds),
+                        "outputTableWriterNodeIds %s must include all tableWriterNodeIds %s",
```

Comment:
nit: "outputTableWriterNodeIds %s must include either all or none of tableWriterNodeIds %s"